cppguide.html

```diff
@@ -4245,29 +4296,57 @@ <h2 id="Naming">Naming</h2>
-understandable by a new reader. Minimize the use of
-abbreviations that would likely be unknown to someone outside
-your project (especially acronyms and initialisms). Do not
-abbreviate by deleting letters within a word. As a rule of thumb,
-an abbreviation is probably OK if it's listed in
- Wikipedia. Generally speaking, descriptiveness should be
-proportional to the name's scope of visibility. For example,
-<code>n</code> may be a fine name within a 5-line function,
-but within the scope of a class, it's likely too vague.</p>
+<p>For the purposes of the naming rules below, a "word" is anything that you
+would write in English without internal spaces. Either words are all lowercase,
+with underscores between words
+("<a href="https://en.wikipedia.org/wiki/Snake_case">snake_case</a>"), or words
+are mixed case with the first letter of each word capitalized
+("<a href="https://en.wikipedia.org/wiki/Camel_case">camelCase</a>" or
+"<a href="https://en.wiktionary.org/wiki/Pascal_case">PascalCase</a>").
+
+</p><h3 id="General_Naming_Rules">Choosing Names</h3>
+
+<p>Give things names that make their purpose or intent understandable to a new
+reader, even someone on a different team than the owners. Do not worry about
+saving horizontal space as it is far more important to make your code
+immediately understandable by a new reader.</p>
+
+<p>Consider the context in which the name will be used. A name should be
+descriptive even if it is used far from the code that makes it available for
+use. However, a name should not distract the reader by repeating information
+that's present in the immediate context. Generally, this means that
+descriptiveness should be proportional to the name's scope of visibility. A free
+function declared in a header should probably mention the header's library,
+while a local variable probably shouldn't explain what function it's within.</p>
+
+<p>Minimize the use of abbreviations that would likely be unknown to someone
+outside your project (especially acronyms and initialisms). Do not abbreviate by
+deleting letters within a word. When an abbreviation is used, prefer to
+capitalize it as a single "word", e.g., <code>StartRpc()</code> rather than
+<code>StartRPC()</code>.</p> As a rule of thumb, an abbreviation is probably OK
+if it's listed in
+
+Wikipedia. Note that certain universally-known abbreviations are OK, such as
+<code>i</code> for a loop index and <code>T</code> for a template
+parameter.
+
+<p>The names you see most frequently are not like most names; a small number of
+"vocabulary" names are reused so widely that they are always in context. These
+names tend to be short or even abbreviated and their full meaning comes from
+explicit long-form documentation rather than from just comments on their
+definition and the words within the names. E.g. <code>absl::Status</code> has a
+dedicated page, with short
+link <a href="https://goto.google.com/status">go/status</a>, documenting its
```

Comment:
This is an internal link. It has a [public equivalent](https://abseil.io/docs/cpp/guides/status), but the fact that it's a short link seems like it might be load-bearing here, so I'm not exactly sure how to fix it. I'll confer with the style team.